### ISSUE TITLE
Insert snippets through shortcodes

### ISSUE THREAD

**Issue by Sophivorus** [sheabunge/code-snippets#58]
I think it would be very useful if we could insert snippets at specific places through shortcodes.

One way to do the UI would be to add a fourth option to the "Reach" section when editing a snippet, after "Execute on all sites", "Execute on admin section" and "Execute on frontend". The fourth option would read "Execute through a shortcode" and when selected, would generate a shortcode like so:

`[code-snippet id=123]`

**Comment by mojenmojen** [comment]
You can already do this. Write your function and use add_shortcode() within the snippet. I use this frequently.

**Comment by Sophivorus** [comment]
Great solution, thanks! I still think this could be a useful and neat feature, but it has much lower priority now. ;-)

**Comment by madivad** [comment]
I've just tried this without success. What have I done wrong?

### in the snippet
    
    add_shortcode('my_shortcode','my_awesome_function');
    function my_awesome_function(){
        echo "AWESOME";
    }

### in the post/page

`[my_shortcode]`

**Comment by mt8** [comment]
Try

```
add_shortcode('my_shortcode','my_awesome_function');
function my_awesome_function(){
    return "AWESOME";
}
```

**Comment by Sophivorus** [comment]
Also, a bit cleaner (in my opinion):
```
add_shortcode( 'my_shortcode', function () {
    return "AWESOME";
});
```

**Comment by madivad** [comment]
It turns out the snippet got deactivated on an error somewhere along the way and I hadn't picked up on it. This is an awesome extension of the plugin! I wish I'd known about this a lot earlier! A generic shortcode through ID would be convenient, even the ability to set a shortcode name (or use the description/slug/custom field) would be convenient, but doing it this way makes it only available if you want it to be.

**Comment by sheabunge** [comment]
I definitely agree that a specialised UI for adding shortcode snippets could be useful. It's sort of already addressed in #28 in the form of HTML content snippets, so I will close this issue in favour of that one, but any additional comments or suggestions are always welcome.
